apps/notifications/src/doc/note-rendering.md

```diff
@@ -0,0 +1,105 @@
+
+```js
+switch ( block.signature.type ) {
+	case 'user':
+		body.push(
+			<User
+				key={ blockKey }
+				block={ block.block }
+				noteType={ this.props.note.type }
+				note={ this.props.note }
+				timestamp={ this.props.note.timestamp }
+				url={ this.props.note.url }
+			/>
+		);
+		break;
+	case 'comment':
+		body.push( <Comment key={ blockKey } block={ block.block } meta={ this.props.note.meta } /> );
+		break;
+	case 'post':
+		body.push( <Post key={ blockKey } block={ block.block } meta={ this.props.note.meta } /> );
+		break;
+	case 'reply':
+		replyBlock = <ReplyBlock key={ blockKey } block={ block.block } />;
+		break;
+	default:
+		body.push( p( html( block.block ) ) );
+		break;
+}
+```
+
+Depending on the item's type, a different block component is rendered - `User` for a representation of a user (for instance, in a notification of a user liking a particular post), `Comment` for a comment made on a blog post, `Post` for a blog post (i.e. on a blog the user is following), `ReplyBlock` for a reply to a comment(?).
+
+_n.b. - these 'blocks' are separate from the concept of 'blocks' in the Gutenberg block editor; a notification might have multiple blocks (for instance, multiple `User` blocks for multiple likes on a post), or it might have a single `Post` block, which then further renders Gutenberg blocks as part of the post content._
+
+These notification block components will be your entry point for any modifications you need to do to the way a particular notification is rendered to the user.
+
+For instance, if the user has received a notification about a post on a blog they are following, the component in question will be `src/panel/notifications/block-post.jsx`:
+
+```js
+import React from 'react';
+
+import { html } from '../indices-to-html';
+import { p } from './functions';
+
+const PostBlock = ( { block } ) => <div className="wpnc__post">{ p( html( block ) ) }</div>;
+
+export default PostBlock;
+```
+
+First, the block is passed to the `html()` function from `/src/panel/indices-to-html/index.js`, which processes the block's content information (making use of the `ranges` mentioned earlier) to split that information into chunks based on how they should be rendered. These chunks are then processed to reintroduce HTML tags back into the plain text content at the points indicated by the values in the `ranges` property.
+
+The result of this function (a string enhanced with HTML tag information) is then passed to `p()` in `src/panel/templates/functions.jsx`, which adds further HTML tag formatting based on line breaks and other special characters (such as backticks for code blocks).
+
+## Troubleshooting and Updating Renderings
+
+Local development of notifications can take place making use of the Calypso dev server, as any changes made to `src/apps/notifications` will be reflected in the Calypso local development environment automatically.
```

Comment:
Changes to the notifications endpoint might be also needed. 

In the particular example of a new post notification that includes blocks, this is the data returned by the API:

<img width="778" alt="Screen Shot 2020-09-10 at 13 58 50" src="https://user-images.githubusercontent.com/1233880/92726652-32a75280-f36e-11ea-9f4e-beb1586dc886.png">

Look at the `text` property. _Heading_ is the content of a `core/heading` block and _Paragraph_ is the content of a `core/paragraph` block. However, the API strips out the `h3` and `p` tags from them, so the components in `src/app/notifications` don't have any info about how to render them correctly [when converting the content to notification blocks](https://github.com/Automattic/wp-calypso/blob/8a700733fbc9564f4fa4d55454d7b40306d26f5c/apps/notifications/src/panel/templates/functions.jsx#L26).

Reply:
Dug into this and found out that the post content is sanitized in `Notification_Block::add_post_block` when running `$block = $this->add_block( $post_content )` (source: `/wp-content/lib/class.notifications-builder.php:1046`).

Not sure if this should be documented here or if the FG page at PCYsg-46P-p2 is enough.

Reply:
I don't see what Miguel is seeing, but perhaps the endpoint that this data is coming from could be documented. This is what I see
<img width="885" alt="Screen Shot 2020-09-10 at 4 41 31 PM" src="https://user-images.githubusercontent.com/66652282/92802570-9f2d4c00-f384-11ea-99fc-000745d03fe8.png">

Reply:
Have you tried expanding the `body` property?  It should be an array of objects that contain the `text` property I was referring to.

Reply:
Yeah -- I think I actually misunderstood your comment. I thought we should expect to see "Heading" and "Paragraph" specifically, but because we have different notifications, I didn't see those specifically. I'm realizing now you were just calling out how they're separated. 👍 
<img width="650" alt="Screen Shot 2020-09-11 at 3 48 06 PM" src="https://user-images.githubusercontent.com/66652282/92967060-98ffa400-f446-11ea-8273-8b566f1ae3ba.png">